### PR TITLE
Circular progress improvement in update area

### LINKED ISSUE #18976: Update spinner appears broken in iOS Home Assistant since 2023.12
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When update is triggered from updates, the usual spinner animation is “broken.”  See video.

https://github.com/home-assistant/frontend/assets/1964829/9682c309-7864-4534-ae60-22994bf590e7

### Describe the behavior you expected

Usual circle spinner animation around component being updated.

### Steps to reproduce the issue

1. Locate updatable item in updates
2. Click into item and select update
3. Click out of dialog and observe spinner animation around item

### What version of Home Assistant Core has the issue?

core-2023.12

### What was the last working version of Home Assistant Core?

core-2023.11.3

### In which browser are you experiencing the issue with?

iOS Home Assistant

### Which operating system are you using to run this browser?

iOS 17.1.2

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

Let me know if this should be an issue in core.

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Circular progress was displayed incorrectly on the update page after the upgrade to material design 3 component. 

For the update area, the UX has been improved while being there: 
* The circular progress now has an aria-label for accessibility. 

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #18976
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/dashboard/ha-config-updates.ts [hunk 1]
```diff
@@ -97,43 +97,46 @@ class HaConfigUpdates extends SubscribeMixin(LitElement) {
...
+                          indeterminate
+                          size="small"
+                          .ariaLabel=${this.hass.localize(
+                            "ui.panel.config.updates.update_in_progress"
+                          )}
+                        ></ha-circular-progress>`
+                      : nothing
+                  }
+                </ha-circular-progress
+              ></span>
```
Comment (bramkragten) [inline src/panels/config/dashboard/ha-config-updates.ts:128]:
```suggestion
                </span>
```

File: src/panels/config/dashboard/ha-config-updates.ts [hunk 2]
```diff
@@ -97,43 +97,46 @@ class HaConfigUpdates extends SubscribeMixin(LitElement) {
...
-              ${!this.narrow
-                ? entity.attributes.in_progress
-                  ? html`<ha-circular-progress
-                      indeterminate
-                      size="small"
-                      slot="meta"
-                    ></ha-circular-progress>`
-                  : html`<ha-icon-next slot="meta"></ha-icon-next>`
-                : ""}
+              <ha-icon-next slot="meta"></ha-icon-next>
```
Comment (bramkragten) [inline src/panels/config/dashboard/ha-config-updates.ts:139]: `hasMeta` is only set when `!this.narrow`, so this will not be shown

**Comment by bramkragten** [comment]
Not sure if I'm a fan of moving the circular progress, especially on mobile space is pretty limited, and placing the it above the image seems ok to me...

Also that the spinner is now no longer in the center of the row bothers me 😅 

@matthiasdebaat any opinion?

**Comment by piitaya** [comment]
Material web is updated so patch is not needed anymore : https://github.com/home-assistant/frontend/pull/19064

**Comment by matthiasdebaat** [comment]
> Not sure if I'm a fan of moving the circular progress, especially on mobile space is pretty limited, and placing the it above the image seems ok to me...
> 
> Also that the spinner is now no longer in the center of the row bothers me 😅
> 
> @matthiasdebaat any opinion?

CleanShot 2023-12-18 at 14 09 58

The current implementation, swapping the chevron with the spinner, is a smart thing to do IMO. Let's keep that.
I would love to keep the dialog and list item animations the same. Can we do something similar as the list item?

**Comment by silamon** [comment]
I've went back to having the circular progress on the state icon. I've considered the design suggestion from Mathias, but the meta icon isn't displayed at all in mobile view and this problem focuses on the smaller view. The list progress can be used but you need to reserve more space and then it's still not clear to what update the list progress belongs. A circular progress isn't clear enough for the when the progress is determinate.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**